java-frontend/src/main/java/org/sonar/java/resolve/FirstPass.java

```diff
@@ -248,6 +250,9 @@ public void visitClass(ClassTree tree) {
       symbol.addTypeParameter((TypeVariableJavaType) typeVariableSymbol.type);
       enterSymbol(typeParameterTree, typeVariableSymbol);
     }
+    if(!tree.typeParameters().isEmpty()) {
```

Comment:
what about extracting `tree.typeParameters()` now that it is called 3 times? (lines 245, 248, 253)